### PR TITLE
Document build config change triggers

### PR DESCRIPTION
**Pull request by csrwng**
Documentation for config change triggers - https://github.com/openshift/origin/pull/4117

### CODE REVIEW

**Comment by csrwng** [comment]
@bparees @adellape ptal

File: dev_guide/builds.adoc [hunk 1]
```diff
@@ -699,6 +699,29 @@ the build that is created uses the `*immutableid*` resolved from the
 `*ImageStream*` referenced by the `*Strategy*`. This ensures that builds
 are performed using consistent image tags for ease of reproduction.
 
+[[config-change-trigger]]
+
+=== Config Change Trigger
+A config change trigger allows a build to be automatically invoked as soon as
+a new `*BuildConfig*` is saved.
```
Comment (bparees) [inline dev_guide/builds.adoc]: s/saved/created/

**Comment by bparees** [comment]
one nit and lgtm.

**Comment by csrwng** [comment]
thx, updated

**Comment by adellape** [comment]
@csrwng Would it be alright to call this a "configuration change trigger" in prose? We try to spell out short-hand "config" in docs when we can, but keep verbatim `*Config*`s when necessary.

**Pull request merged**